### DISCUSSION TITLE
preserve partially planned result for next task

### DISCUSSION THREAD

**Discussion by maxware0715** [frePPLe/frepple#485] [Q&A] [answered]
I try to run a new output planning task with result of sales orders (SO) and manufacturing orders (MO) from previous round. In the new round, items, operations related tables will be updated while features related to SO and MO are not affected.

I try to solve this problem in DB base, so solutions in psql are preferred than in interface   
layer.

During the process, I met several challenges:

1. Since I need to mark each wip in my task, I put them in MO with confirm state to name them by reference rather then put them in buffer. After planning task, some SO and MO are partially pegged. Is there any solution to keep the pegging result for next round?

I prefer this can be solved by adding features in DB tables because I need to reload SO and MO after update operations.

2. If 1 is not possible, I try to split pegged part of the result. Besides remained SO, I can not calculate remained wips (MO) in an intuitive way. The most approach information is inventory  detail and reports, but they are number of wips affected by start/end ratio in operation materials. Any methods to calculate remained wips?

3. I try to add up to 16 cpu for acceleration, but the cpu rate is only 400% rather than 1600% in my environment. Is there any bottleneck of frepple itself?

**Comment by jdetaeye** [comment, marked as answer]
>>1. Since I need to mark each wip in my task, I put them in MO with confirm state to name them by reference rather then put them in buffer. After planning task, some SO and MO are partially pegged. Is there any solution to keep the pegging result for next round?

We can do "dynamic pegging", which is the default. With every planning run the pegging is recomputed based on a FIFO material allocation in the material buffer.  

We can do "hard pegging" where you can use the batch field to create a hard reservation of an MO to an SO. 

We haven't done any project yet where we switched between both types in different planning runs.  (And personally, I think it gets very messy...)

>> 2. If 1 is not possible, I try to split pegged part of the result. Besides remained SO, I can not calculate remained wips (MO) in an intuitive way. The most approach information is inventory detail and reports, but they are number of wips affected by start/end ratio in operation materials. Any methods to calculate remained wips?

I don't understand the question.

>> 3. I try to add up to 16 cpu for acceleration, but the cpu rate is only 400% rather than 1600% in my environment. Is there any bottleneck of frepple itself?

Frepple partitions the planning problem in isolated sub-problems.   If we can only identify 4 sub-problems, then we'll use only 4 planning threads.

**Reply by maxware0715** [reply]
Thank you.
The second problem is how to get the remain quantity of my input MO after pegging? Demand only shows the unit of end items, which is different from MO item unit.
